build.js

```diff
@@ -1824,10 +1861,26 @@ var dirtyDetectingBuilder = function(rootPackageName, resultsCache, prevResultsC
     var compileModulesCommands =
       singleFileCompile + compileModuleOutputs.map( function(c){return c.compileOutputString;}).join(' ');
 
+    var compileForeignDependencies = null;
+    if (commonML.foreignDependencies.length > 0) {
+      // We cd into the main directory of the library, then run the C build
+      // command, then cd back to the user's library directory. This seems to be
+      // the only way to control where the output .o file should be generated.
+      compileForeignDependencies =
+        ['cd', packageResource.realPath, '&&']
+        .concat(programForCompiler(buildConfig.compiler))
+        .concat(['-c'])
```

Comment:
But where does this output go? If you have:

`A` dependsOn `B`
`C` dependsOn `B`

When building `A` from the top, all of `B`'s artifacts should go in `A/_build/B/` and when building `C` from the top, all of `B`'s artifacts should go in `C/_build/B`.

Both `A` and `B` could symlink to `B` so we don't want their building of `B`'s artifacts to collide.

We have one exception to this rule for the .merlin file - but I would even like to get rid of that special case.
Do you know of a way to control the output location?